File: KerningTest/ViewController.m
//
//  ViewController.m
//  KerningTest
//
//  Created by Oliver Drobnik on 6/16/13.
//  Copyright (c) 2013 Oliver Drobnik. All rights reserved.
//

#import "ViewController.h"

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
	
	
	NSDictionary *attributes = @{NSFontAttributeName: [UIFont systemFontOfSize:30]};
	
	NSMutableAttributedString *attributedString = [[NSMutableAttributedString alloc] initWithString:@"Kerning Examples:\n\nTw\nAV\nWA\nWa\n\nThe second character should always be visible under some part of the first." attributes:attributes];
	
	self.textView.attributedText = attributedString;
}

@end

File: ImageLinkCrash/ImageLinkCrash/ViewController.swift
//
//  ViewController.swift
//  ImageLinkCrash
//
//  Created by Stefan Gugarel on 28/07/2017.
//  Copyright © 2017 Stefan Gugarel. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
	
	@IBOutlet weak var textView: UITextView!
	
	override func viewDidLoad() {
		
		super.viewDidLoad()
		
		// Do any additional setup after loading the view, typically from a nib.
		let image = UIImage(named: "ScalpelSi")
		
		// Image attachment
		let textAttachment = NSTextAttachment()
		textAttachment.image = image
		
		// Create attributed string
		let attributedString = NSAttributedString(attachment: textAttachment)
		let mutableString = NSMutableAttributedString(attributedString: attributedString)

		// Add Link
		mutableString.addAttributes([NSLinkAttributeName : "http://www.cannondale.com/"], range: NSMakeRange(0, mutableString.length))
		
		textView.attributedText = mutableString
		textView.isSelectable = true
	}
}

